lib/TPP/IR/MatcherUtils.cpp

```diff
@@ -151,7 +144,82 @@ static bool hasReluBody(Operation *op, SmallVectorImpl<Value> *captured) {
     }
     return false;
   };
-  return (getOperand(maxfLhs, maxfRhs) || getOperand(maxfRhs, maxfLhs));
+
+  // Multiple patterns map to Relu.
+  if (isa<arith::MaximumFOp>(innerOp)) {
+    // Pattern 1 - arith.maximumf(tensor, const 0)
+    if (yieldOp->getOperand(0).getDefiningOp() != innerOp)
+      return false;
+    auto maxfOp = cast<arith::MaximumFOp>(innerOp);
+    Value maxfLhs = maxfOp.getLhs();
+    Value maxfRhs = maxfOp.getRhs();
+
+    return (getOperand(maxfLhs, maxfRhs) || getOperand(maxfRhs, maxfLhs));
+  } else if (isa<arith::CmpFOp>(innerOp)) {
+    // Pattern 2 - arith.cmpf, arith.select
+    // %22 = arith.cmpf ugt, %in, %cst : f32
```

Comment:
I would drop this IR.

Reply:
I was keeping it as a doc for easier code understanding.

Reply:
Reworded it to make it more abstract.